deepview_profile/commands/analysis.py

```diff
@@ -0,0 +1,165 @@
+    parser.set_defaults(func=main)
+
+def measure_breakdown(session, nvml):
+    print("analysis: running measure_breakdown()")
+    yield session.measure_breakdown(nvml)
+    release_memory()
+
+def measure_throughput(session):
+    print("analysis: running measure_throughput()")
+    yield session.measure_throughput()
+    release_memory()
+
+def habitat_predict(session):
+    print("analysis: running deepview_predict()")
+    yield session.habitat_predict()
+    release_memory()
+
+def measure_utilization(session):
+    print("analysis: running measure_utilization()")
+    yield session.measure_utilization()
+    release_memory()
+
+def energy_compute(session):
+    print("analysis: running energy_compute()")
+    yield session.energy_compute()
+    release_memory()
+
+def hardware_information(nvml):
+    
+    hardware_info = { 
+        'hostname': platform.node(),
+        'os': " ".join(list(platform.uname())),
+        'gpus': nvml.get_device_names()
+    }
+    return hardware_info
+
+def actual_main(args): 
+    from deepview_profile.analysis.session import AnalysisSession
+    from deepview_profile.exceptions import AnalysisError
+
+    if os.path.exists(args.output):
+        print(
+            "ERROR: The specified output file already exists.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    try: 
+        project_root = os.getcwd()
+        data = { 
+            "analysisState": {
+                "message_type": "analysis",
+                "project_root": project_root,
+                "project_entry_point": args.entry_point,
+                "hardware_info": {},
+                "throughput": {},
+                "breakdown": {},
+                "habitat": {},
+                "additionalProviders": "",
+                "energy": {},
+                "utilization": {}
+            }
+        }
+
+        session = AnalysisSession.new_from(project_root, args.entry_point)
+        release_memory()
+
+        is_return_all = args.all
+
+        with NVML() as nvml:
+            data['analysisState']['hardware_info'] = hardware_information(nvml)
+            if args.measure_breakdown or is_return_all:
+                data['analysisState']['breakdown'] = next_message_to_dict(measure_breakdown(session, nvml))
+
+        if args.measure_throughput or is_return_all:
+            data['analysisState']['throughput'] = next_message_to_dict(measure_throughput(session))
+
+        if args.habitat_predict or is_return_all:
+            data['analysisState']['habitat'] = next_message_to_dict(habitat_predict(session))
+
+        if args.measure_utilization or is_return_all:
+            data['analysisState']['utilization'] = next_message_to_dict(measure_utilization(session))
+
+        if args.energy_compute or is_return_all:
+            data['analysisState']['energy'] = next_message_to_dict(energy_compute(session))
+
+        with open(args.output, "w") as json_file:
+            json.dump(data, json_file, indent=4)

```

Comment:
For this to work with the web viewer, we must also include the source files (i.e. model implementation). 

The source files are stored in a field called `fileContents` and is encoded with base64. You can find a reference implementation here: https://github.com/CentML/DeepView.Explore/blob/main/deepview-explore/src/skyline_session.ts#L218-L233

Reply:
Please, check our version of `encodedFiles` section. 

It was a bit confusing, the files from DeepView.Explorer, probably, just used for breakdown section and for our implementation, there are possible other optional sections. For now, we are adding all files, except of entry_point.py file to the list of encodedFiles. Please, approve if it is correct?

Thank you very much.